app/controllers/api/v1/tags_controller.rb

```diff
@@ -0,0 +1,11 @@
+class Api::V1::TagsController < ApplicationController
+
+  def index
+    if params[:recipe_id]
+      @tags = Recipe.find_by!(id: params[:recipe_id]).tags
+    else
+      @tags = Tag.all
+    end
+    render json: @tags
+  end
+end
```

Comment:
Can you go into your VSCode settings and enable final newlines? From VS Code, press `Command-,`, then click on `Text Editor`, then `Files`. Scroll down a bit and check the box under `Insert Final Newline`.

It's good practice to have a newline at the end of every file, as it's a standard, and certain editors and parsers can choke on the lack of them. To fix that, we'll need to add a blank line at the end of this file, but with that VSCode setting, it'll do it automatically when you save the file.